Content.Client/Atmos/Visualizers/PortableScrubberVisualsSystem.cs

```diff
@@ -0,0 +1,32 @@
+using Robust.Client.GameObjects;
+using Content.Shared.Atmos.Visuals;
+using Content.Client.Power;
+
+namespace Content.Client.Atmos.Visualizers
+{
+    /// <summary>
+    /// Controls client-side visuals for portable scrubbers.
+    /// </summary>
+    public sealed class PortableScrubberSystem : VisualizerSystem<PortableScrubberVisualsComponent>
+    {
+        protected override void OnAppearanceChange(EntityUid uid, PortableScrubberVisualsComponent component, ref AppearanceChangeEvent args)
+        {
+            if (args.Sprite == null)
+                return;
+
+            if (args.Component.TryGetData(PortableScrubberVisuals.IsFull, out bool isFull)
+                && args.Component.TryGetData(PortableScrubberVisuals.IsRunning, out bool isRunning))
+            {
+                var runningState = isRunning ? component.RunningState : component.IdleState;
+                args.Sprite.LayerSetState(PortableScrubberVisualLayers.IsRunning, runningState);
+
+                var fullState = isFull ? component.FullState : component.ReadyState;
+                args.Sprite.LayerSetState(PowerDeviceVisualLayers.Powered, fullState);
+            }
+        }
+    }
```

Comment:
this looks easy to convert entirely into a generic visualizer, the other atmos ones are so you can look at those, and i think the visualizer conversion docs have notes on it

Reply:
Couldn't get it working